### PR TITLE
PanelHeaderCorner: Remove font-awesome icons

### LINKED ISSUE #59592: Design System: Continue replacing Font-Awesome icons with SVGs
**What happened**:
The "i" for the description (table-panel) disappear, when using this HTML-Script inside of a "Text"/"HTML"-Panel:
image
What "i" do I mean:
image
image

**What you expected to happen**:
The "i" is still visible!

**How to reproduce it (as minimally and precisely as possible)**:
Create a KIT on Font-Awesome and copy it to a html-panel.
[https://fontawesome.com/kits](https://fontawesome.com/kits)
Add a description to a e.g. table panel, save it and the dashboard. Reload the page if necessary.
This will occour:
image
image

**Anything else we need to know?**:
I guess no. Please let ME know, if you need anything!

**Environment**:
- Grafana version: 9.2.5
- Data source type & version: irrelevant
- OS Grafana is installed on: linux
- User OS & Browser: Windows & Opera/Chorme/Edge/...
- Grafana plugins: irrelevant
- Others: irrelevant

### PR DESCRIPTION
**Pull request by Clarity-89**
**What is this feature?**
- Replaces font-awesome icons from `PanelHeaderCorner` with the `Icon` component 
- Replaces related SCSS with Emotion styles.

**Why do we need this feature?**

To get rid of the FA icons. 

**Which issue(s) does this PR fix?**:

Fixes https://github.com/grafana/grafana/issues/59592

**Special notes for your reviewer:**

**Before**:
Screenshot 2024-05-03 at 13 00 41
Screenshot 2024-05-03 at 14 33 42
Screenshot 2024-05-03 at 14 34 09

**After**
Screenshot 2024-05-03 at 14 39 02
Screenshot 2024-05-03 at 14 33 24
Screenshot 2024-05-03 at 14 39 41

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: public/app/features/dashboard/components/PanelEditor/PanelHeaderCorner.tsx [hunk 1]
```diff
@@ -103,15 +91,86 @@ export class PanelHeaderCorner extends Component<Props> {
...
+  const ariaLabel = selectors.components.Panels.Panel.headerCornerInfo(infoMode.toLowerCase());
+  const styles = useStyles2(getStyles);
+
+  return (
+    <Tooltip content={content} placement="top-start" theme={theme} interactive>
+      <button type="button" className={styles.infoCorner} onClick={onClick} aria-label={ariaLabel}>
+        <Icon
+          name={iconMap[infoMode]}
+          size={infoMode === InfoMode.Links ? 'sm' : 'lg'}
+          className={cx(styles.icon, infoMode === InfoMode.Links ? styles.iconLinks : '')}
```
Comment (ashharrison90) [inline public/app/features/dashboard/components/PanelEditor/PanelHeaderCorner.tsx:124]: proper way to conditionally apply styles using `cx` would be something like:
```suggestion
          className={cx(styles.icon, {
            [styles.iconLinks]: infoMode === InfoMode.Links
          })}
```

same applies a couple of lines below

File: public/app/features/dashboard/components/PanelEditor/PanelHeaderCorner.tsx [hunk 2]
```diff
@@ -103,15 +91,86 @@ export class PanelHeaderCorner extends Component<Props> {
...
+  [InfoMode.Links]: 'external-link-alt',
+};
+
+const getStyles = (theme: GrafanaTheme2) => {
+  return {
+    icon: css({
+      position: 'absolute',
+      top: 0,
+      left: 0,
+      zIndex: 3,
```
Comment (ashharrison90) [inline public/app/features/dashboard/components/PanelEditor/PanelHeaderCorner.tsx:144]: gotta ask about the magic `zIndex` numbers

in the previous scss, i believe this resolved to 2 - why 3 now? 🤔

File: public/app/features/dashboard/components/PanelEditor/PanelHeaderCorner.tsx [hunk 3]
```diff
@@ -103,15 +91,86 @@ export class PanelHeaderCorner extends Component<Props> {
...
+      background: 'none',
+      border: 'none',
+      color: theme.colors.text.secondary,
+      cursor: 'pointer',
+      position: 'absolute',
+      left: 0,
+      top: 0,
+      width: theme.spacing(4),
+      height: theme.spacing(4),
+      zIndex: 11,
```
Comment (ashharrison90) [inline public/app/features/dashboard/components/PanelEditor/PanelHeaderCorner.tsx:173]: same question here, why 11 and not 3 as before?

**Review by ashharrison90 — COMMENTED** [review]
generally looks great, couple of nits

File: public/app/features/dashboard/components/PanelEditor/PanelHeaderCorner.tsx (hunk 3 above)
Comment (Clarity-89) [reply public/app/features/dashboard/components/PanelEditor/PanelHeaderCorner.tsx:173]: Good catch, I've misread the sass variable 🙈

**Review by ashharrison90 — APPROVED** [review]
lgtm 👍

**Pull request merged**